test/contracts/staking/staking-factory.test.ts

```diff
@@ -42,26 +62,187 @@ export type StakingPoolLibData = {
+            })
+            after(async () => {
+                await stakingPoolFactory.unpause()
+                await stakingPoolFactory.updateTokenSweepBeneficiary(treasury)
+            })
+        })
+
+        describe('ERC20 token sweep', () => {
+            it('side effects', async () => {
+                const seedFunds = 100n
+                const sweepAmount = 55n
+                await successfulTransaction(
+                    collateralTokens.transfer(
+                        stakingPoolFactory.address,
+                        seedFunds
+                    )
+                )
+                expect(
+                    await collateralTokens.balanceOf(stakingPoolFactory.address)
+                ).equals(seedFunds)
+                expect(await collateralTokens.balanceOf(treasury)).equals(0)
+
+                const receipt = await successfulTransaction(
+                    stakingPoolFactory.sweepERC20Tokens(
+                        collateralTokens.address,
+                        sweepAmount
+                    )
+                )
+
+                expect(
+                    await collateralTokens.balanceOf(stakingPoolFactory.address)
+                ).equals(seedFunds - sweepAmount)
+                expect(await collateralTokens.balanceOf(treasury)).equals(
+                    sweepAmount
+                )
+                const expectedEvents: ExpectedERC20SweepEvent[] = [
+                    {
+                        beneficiary: treasury,
+                        tokens: collateralTokens.address,
+                        amount: sweepAmount,
+                        instigator: admin
+                    }
+                ]
+                verifyERC20SweepEvents(receipt, expectedEvents)
+                verifyERC20SweepLogEvents(
+                    stakingPoolFactory,
+                    receipt,
+                    expectedEvents
+                )
+            })
+
+            it('only Super User', async () => {
+                await expect(
+                    stakingPoolFactory
+                        .connect(nonAdmin)
+                        .sweepERC20Tokens(collateralTokens.address, 5)
+                ).to.be.revertedWith(
+                    accessControlRevertMessageMissingGlobalRole(
+                        nonAdmin,
+                        SUPER_USER
+                    )
+                )
+            })
+
+            it('only when not paused', async () => {
+                await stakingPoolFactory.pause()
+
+                await expect(
+                    stakingPoolFactory.sweepERC20Tokens(
+                        collateralTokens.address,
+                        5
+                    )
+                ).to.be.revertedWith('Pausable: paused')
+            })
+        })
+        after(async () => {
+            await stakingPoolFactory.unpause()
+        })
     })
 
     describe('create pools', () => {
+        before(async () => {
```

Comment:
question: is the `epochStartTimestamp` needed in this changeset or a general fix?

Reply:
Actually increasing the START_DELAY var negates the need to do this. I guess each hardhat transaction rolls the chain timestamp forward and by the time these tests were being run, the initial timestamp had be surpassed (and the contracts disallow historic timestamps for the creation of pools).